images/handlers.go

```diff
@@ -147,28 +148,52 @@ func WalkNotEmpty(ctx context.Context, handler Handler, descs ...ocispec.Descrip
 // A concurrency limiter can be passed in to limit the number of concurrent
 // handlers running. When limiter is nil, there is no limit.
 //
+// A duplicate suppression can be used to cooperate with other Dispatch to
+// limit duplicate handlers on the same descriptor.
+//
 // Typically, this function will be used with `FetchHandler`, often composed
 // with other handlers.
 //
 // If any handler returns an error, the dispatch session will be canceled.
-func Dispatch(ctx context.Context, handler Handler, limiter *semaphore.Weighted, descs ...ocispec.Descriptor) error {
+func Dispatch(
+	ctx context.Context,
+	handler Handler,
+	limiter pkgsemaphore.Semaphore,
+	duplicateSuppression pkgsingleflight.Singleflighter,
+	descs ...ocispec.Descriptor,
+) error {
```

Comment:
Can we avoid breaking Go API?
Let’s have `DispatchWithOpts(Context, Handler, Semaphore, *DispatchOpts, …Descriptor)`

Reply:
Nice! Thanks for the suggestion.